module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php

```diff
@@ -1170,4 +1170,112 @@ public function getTextualHoldings()
+     *
+     * @param array $marc_data MARC data for a specific field
+     * @param array $indData   Array containing the indicator number as the key
+     * and the value as an array of strings for the allowed indicator values
+     * ex: [['1' => ['1', '2']], ['2' => ['']]] would filter fields ind1 = 1 or 2 or ind2 = blank
+     * ex: [['1' => ['1'], '2' => ['7']]] would filter fields with ind1 = 1 and ind2 = 7
+     * ex: [] would apply no filtering based on indicators
+     *
+     * @return bool
+     */
+    protected function checkIndicatorFilters($marc_data, $indData): bool
+    {
+        foreach ($indData as $indFilter) {
+            if ($this->checkIndicatorFilter($marc_data, $indFilter)) {
+                return true;
+            }
+        }
+        // If we got this far, either $indData is empty (no filters defined -- return true)
+        // or it is non-empty (no filters matched -- return false)
+        return empty($indData);
+    }
+
+    /**
+     * Takes a Marc field that notes are stored in (ex: 950) and a list of
+     * sub fields (ex: ['a','b']) optionally as well as what indicator
+     * numbers and values to filter for and concatenates the subfields
+     * together and returns the fields back as an array
+     * (ex: ['subA subB subC', 'field2SubA field2SubB'])
+     *
+     * @param string $field    Marc field to search within
+     * @param ?array $subfield Sub-fields to return or empty for all
+     * @param array  $indData  Array containing the indicator number as the key
+     * and the value as an array of strings for the allowed indicator values
+     * ex: [['1' => ['1', '2']], ['2' => ['']]] would filter fields ind1 = 1 or 2 or ind2 = blank
+     * ex: [['1' => ['1'], '2' => ['7']]] would filter fields with ind1 = 1 and ind2 = 7
+     * ex: [] would apply no filtering based on indicators
+     *
+     * @return array The values within the subfields under the field
+     */
+    public function getMarcFieldWithInd(
+        string $field,
+        ?array $subfield = null,
+        array $indData = []
+    ) {
+        $vals = [];
+        $marc = $this->getMarcReader();
+        $marc_fields = $marc->getFields($field, $subfield);
+        foreach ($marc_fields as $marc_data) {
+            $field_vals = [];
+            if ($this->checkIndicatorFilters($marc_data, $indData)) {
+                $subfields = $marc_data['subfields'];
+                foreach ($subfields as $subfield) {
+                    $field_vals[] = $subfield['data'];
+                }
+            }
+            $newVal = implode(' ', $field_vals);
+            if (!empty($field_vals) && !in_array($newVal, $vals)) {
+                $vals[] = $newVal;
+            }
+        }
+        return $vals;
+    }
+
+    /**
+     * Get the abstract and summary notes
+     *
+     * @return array Note fields from the MARC record
+     */
+    public function getAbstractAndSummaryNotes()
```

Comment:
Since we're adding these new methods, it would probably be a good idea to add them to the default RecordDataFormatter configuration so they can actually be used in places visible to the user as well as the data export -- I see no harm in having more visible notes in the description tab. That's probably out of scope for this PR, but maybe we can do that as a separate follow-up PR when this one is merged. What do you think?

Reply:
I think that idea, but we probably should make it a separate PR since it does raise the question on if we would want them separated on the display. For example, in [our customization](https://github.com/MSU-Libraries/catalog/blob/main/vufind/module/Catalog/src/Catalog/RecordDriver/SolrMarc.php#L272-L326) we have the 520 field separated based on the indicator so they have different appropriate labels. For the RIS export it made sense to combine abstract and summary (as AB is defined as "Abstract or synopsis"), but we probably want them separate on the description tab.

If I make a PR after this one is done to add it to the display, do you have a preference on having the 520 split to separate fields like we have it or do you think just have it use what is in the new `getAbstractAndSummaryNotes` function? (If we split them, I think I'd update `getAbstractAndSummaryNotes` to call `getSummaryNotes` and `getAbstractNotes` to avoid duplication of logic).

Reply:
I think it makes sense to do it in a more granular way and then make getAbstractAndSummaryNotes a wrapper as you propose. You could feel free to make the record driver changes here if you like!

Reply:
Unless you object I think I'd prefer making a separate PR since this one is focused on RIS export and not display. Then the new PR could include adding all the separate 520 fields to the description tab (and refactoring `getAbstractAndSummaryNotes` as described earlier). I can work on creating that one after this one is merged.

Reply:
If that's how you prefer to do things, I don't object... but one other thought in case you like it: what if we drop getAbstractAndSummaryNotes entirely, and just use getSummaryNotes and getAbstractNotes in the RIS template? Then if another record driver supports only one or the other, the whole thing will still work... and we end up needing/using fewer methods.

Reply:
Ok I added them to the display as part of this PR. I updated `getSummary` to `return $this->getMarcFieldWithInd('520', ['a'], [[1 => ['', '0', '2', '8']]]);` instead of `return $this->getFieldArray('520');`  to prevent the other separate 520 fields from being duplicated in there as well as their respective fields. The only reason I'm not having `getSummary` call `getSummaryNotes` was because I didn't think we'd want to affect the existing display that only displays the `a` subfield when calling `getSummary`.

Reply:
That makes sense. A couple questions/comments, though:

1.) Is it possible we at least want to limit some of these to the alphabetic rather than numeric subfields? I'm not sure that we want to include things like URIs in raw user output or RIS export.

2.) I was suggesting adjusting the record driver methods here but leaving the actual display changes for a separate PR -- mainly because if we add new note fields to the display we also need to add the labels to the language files, and right now, we're in a language file freeze because release 10.1 is very close. If all of this work can wait to be merged until release 11.0, that's no problem... but if you wanted to try to get some of this into 10.1, we should split off any parts that require new language strings. Sorry for being a bit unclear about my intention before! :-)

Reply:
Sounds good. I took out the changes to `RecordDataFormatterFactory.php` and the 520 fields not used in the RIS export -- we can add them back in later.

Agreed about the alphabetic subfields -- I made that update and updated the test fixture to make sure it gets excluded.